lib/queryRelatedRecords.js

```diff
@@ -0,0 +1,61 @@
+const { getCollectionCrs, getGeometryTypeFromGeojson } = require('./helpers')
+const { renderStats } = require('./templates')
+const _ = require('lodash')
+const { computeFieldObject } = require('./field')
+
+module.exports = queryRelatedRecords
+
+function queryRelatedRecords (data, params = {}) {
+  // TODO clean up this series of if statements
+  const filtersApplied = data.filtersApplied || {}
+  const options = _.cloneDeep(params)
+
+  if (filtersApplied.projection || options.returnGeometry === false) delete options.outSR
+  if (filtersApplied.geometry) delete options.geometry
+  if (filtersApplied.where || options.where === '1=1') delete options.where
+  if (filtersApplied.offset) delete options.resultOffset
+  if (filtersApplied.limit) {
+    delete options.resultRecordCount
+    delete options.limit
+  }
+  if (data.statistics) return renderStats(data)
+  if (options.returnCountOnly && data.count !== undefined) return { count: data.count }
+  if (options.f !== 'geojson' && !options.returnExtentOnly) options.toEsri = true
```

Comment:
Written this way to match the structure of the query route code for familiarity when maintaining, certainly can be simplified.

Reply:
In query.js the `options` is built up so it can be used later on in the handler, i.e., it is passed on into other function calls. See [here](https://github.com/koopjs/FeatureServer/blob/master/lib/query.js#L36-L45). In [queryRelatedRecords.js](https://github.com/koopjs/FeatureServer/pull/209/files/0167585e77bceb0964047eaac2d39c5877750eee#diff-815ab5f07ca6dc69ecbbc49972ef1137db03f0fc61c8c0d108a5d9315b90e093), the `options` is never passed on to anything. Given that, it seems like this [block of code](https://github.com/koopjs/FeatureServer/blob/0167585e77bceb0964047eaac2d39c5877750eee/lib/queryRelatedRecords.js#L9-L23) has been unnecessarily added to this handler.  I'd say that goes for [this line as well](https://github.com/koopjs/FeatureServer/blob/0167585e77bceb0964047eaac2d39c5877750eee/lib/queryRelatedRecords.js#L21) because the [`queryRelatedRecords` method](https://developers.arcgis.com/rest/services-reference/enterprise/query-related-records-feature-service-.htm) makes no mention of supporting statistics.